src/fuselage/CCPACSFuselages.cpp

```diff
@@ -90,4 +93,75 @@ int CCPACSFuselages::GetFuselageIndex(const std::string& UID) const
+                         fuselageUID + "\". This uid is already present in the file. Choose another uid.");
+    }
+
+    // check if the profile uid is valid
+    try {
+        uidManager.ResolveObject<CCPACSFuselageProfile>(profileUID);
+    }
+    catch (const CTiglError& e) {
+        throw CTiglError("CCPACSFuselages::CreateFuselage: Impossible to create the fuselage with the profile uid \"" +
+                         profileUID +
+                         "\". This uid seems not to be present or to not reference a profile. Make sure to use a "
+                         "fuselage profile UID present in the file.");
+    }
+
+    CCPACSFuselage& fuselage = AddFuselage();
+    fuselage.SetUID(fuselageUID);
+    fuselage.SetName(fuselageUID);
+    fuselage.GetTransformation().Init(uidManager.MakeUIDUnique(fuselageUID + "Transformation"));
+    boost::optional<std::string> description("Fuselage generated from the function CCPACSFuselages::CreateFuselage");
+    fuselage.SetDescription(description);
+
+    // Create section and element
+    std::string tempSectionUID;
+    std::string tempElementUID;
+    CTiglTransformation tempElementTransformation;
+    tempElementTransformation.SetIdentity();
+    for (int i = 1; i <= numberOfSection; i++) {
+
+        tigl::CCPACSFuselageSection& section = fuselage.GetSections().AddSection();
+        tempSectionUID                       = uidManager.MakeUIDUnique(fuselageUID + "Sec" + std::to_string(i));
+        section.SetUID(tempSectionUID);
+        section.SetName(tempSectionUID);
+        section.GetTransformation().Init(uidManager.MakeUIDUnique(tempSectionUID + "Tr"));
+
+        tigl::CCPACSFuselageSectionElement& element = section.GetElements().AddElement();
+        tempElementUID                              = uidManager.MakeUIDUnique(tempSectionUID + "Elem1");
+        element.SetUID(tempElementUID);
+        element.SetName(tempElementUID);
+        element.GetTransformation().Init(uidManager.MakeUIDUnique(tempElementUID + "Tr"));
+        element.GetTransformation().setTransformationMatrix(tempElementTransformation);
+        element.SetProfileUID(profileUID);
+
+        // increment the position of the element
+        tempElementTransformation.AddTranslation(1, 0, 0);
+    }
+
+    // Create segment
+    for (int i = 1; i < fuselage.GetSections().GetSectionCount(); i++) {
+        CCPACSFuselageSectionElement& fromElement = fuselage.GetSection(i).GetSectionElement(1);
+        CCPACSFuselageSectionElement& toElement   = fuselage.GetSection(i + 1).GetSectionElement(1);
+        CCPACSFuselageSegment& segment            = fuselage.GetSegments().AddSegment();
+        std::string segmentUID                    = uidManager.MakeUIDUnique(fuselageUID + "Seg" + std::to_string(i));
+        segment.SetUID(segmentUID);
+        segment.SetName(segmentUID);
+        segment.SetFromElementUID(fromElement.GetUID());
+        segment.SetToElementUID(toElement.GetUID());
+    }
+    return fuselage;
```

Comment:
Why is the fuselage not added to the fuselages list?

Why does this function then need to be part of the fuselages vector?

Could'nt this be a standalone factory function?

Reply:
Why is the fuselage not added to the fuselages list?
    -> The fuselage is added in fuselage list by the call of " CPACSFuselages::AddFuselage()" in the function. 

Could'nt this be a standalone factory function?
   -> Since the fuselage is register  and change the object itself seems hard to make a standalone factory.